### PR TITLE
93 2 step new

### PR DESCRIPTION
**Pull request by Wolfgang1208**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**
1, Created a form for new users to fill in details
2, Redirect if users' doc on firestore having any null values in name, email, phone.

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

**Comment by skyheat** [comment]
396B5272-3948-4E83-A279-20EF61F6E486

page is bugged on phone

**Review by skyheat — CHANGES_REQUESTED** [review]
Looks great otherwise! Good job 🥇

File: src/pages/signupDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { NextPage } from 'next'
+import Image from 'next/image'
+
+import NavBar from '@/components/NavBar'
+// import LoginPanel from '@/components/Login/LoginPanel'
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/pages/signupDetails.tsx:5]: The image is used for poops logo

File: src/components/UpdateDetails/UpdateDetailsPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    recaptchaVerifier: RecaptchaVerifier
+    confirmationResult: ConfirmationResult
+  }
+}
+
+import { useRouter } from 'next/router'
+
+import SubmitButton from '@/components/UpdateDetails/SubmitButton'
+
+// TODO add correct redirect to the pages path // done
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/components/UpdateDetails/UpdateDetailsPanel.tsx:25]: yes

File: src/components/UpdateDetails/UpdateDetailsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+          } catch (error) {
+            // console.log(error)
+          }
+          handleDocUpdate(currentUser)
+          updateProfile(currentUser, { displayName: displayName })
+          setDone(true)
+        }
+      }
+    } catch (error) {
+      setErr(true) // debug only
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/components/UpdateDetails/UpdateDetailsPanel.tsx:100]: removed

File: src/components/UpdateDetails/UpdateDetailsPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          try {
+            window.recaptchaVerifier = new RecaptchaVerifier(
+              'recaptcha-container',
+              {
+                size: 'invisible'
+              },
+              auth
+            )
+          } catch (error) {
+            // console.log(error)
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/components/UpdateDetails/UpdateDetailsPanel.tsx:92]: removed

File: src/components/UpdateDetails/UpdateDetailsPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  // push test
+  const handleSubmit = (
+    e: React.MouseEvent<HTMLButtonElement>,
+    contact: Contact | null,
+    currentUser: User
+  ) => {
+    e.preventDefault()
+
+    try {
+      setErr(false) // debug only
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/components/UpdateDetails/UpdateDetailsPanel.tsx:77]: removed

File: src/components/UpdateDetails/UpdateDetailsPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+  const handleDocUpdate = async (currentUser: User) => {
+    const userDocRef = doc(db, 'users', currentUser.uid)
+    await updateDoc(userDocRef, {
+      'info.email': email,
+      'info.name': displayName,
+      'info.phone': phoneNumber
+    })
+  }
+  // push test
```
[the remark this answers is not in the document]
Comment (Wolfgang1208) [reply src/components/UpdateDetails/UpdateDetailsPanel.tsx:68]: removed

**Comment by Wolfgang1208** [comment]
request changes done

**Comment by Wolfgang1208** [comment]
Checked 2-step login in code works well after applied Jun's changes.

File: src/pages/signupDetails.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (skyheat) [reply src/pages/signupDetails.tsx]: I accidently selected lines 2-4, should only be removing line 5

**Review by skyheat — APPROVED** [review]

**Pull request merged**